zmoney/Controllers/TransactionsViewController.swift

```diff
@@ -17,16 +17,24 @@ class TransactionsViewController: UIViewController {
 
     override func viewDidLoad() {
         super.viewDidLoad()
-        stateController = StateManager(rootView: self.view, loadedView: tableView)
+
+        if let windowScene = UIApplication.shared.connectedScenes.first as? UIWindowScene,
+           let sceneDelegate = windowScene.delegate as? SceneDelegate {
+            sceneDelegate.applicationCoordinator.delegate = self
+        }
+
```

Comment:
Решение через делегата в координаторе переуплотняет сильно схему. И если, к примеру, нам нужно будет в каком-то другом месте приложения отслеживать статус авторизации (в кнопке какой-нибудь), то придется ее связывать с ApplicationCoordinator.delegate, а делегат только один может быть и т.п.

Здесь лучше использование паттерна Observer подойдет. И я предлагаю сделать оповещения через NotificationCenter. Будет тогда как-то так

```suggestion
NotificationCenter.default.addObserver(forName: .zMoneyConfigUpdated, object: nil, queue: nil) { (_) in
            self.refreshTransactionsList()
        }
```

И все, остальную обвязку с делегатом можно удалить.